miner/worker.go

```diff
@@ -92,7 +92,7 @@ type Result struct {
 type worker struct {
 	mu sync.Mutex
 
-	agents []Agent
+	agents map[Agent]struct{}
```

Comment:
Please elaborate what the benefit of being able to call `delete` is, compared to the current code which already has a working implementation of deleting an element from a slice.